src/App.tsx

```diff
@@ -3,33 +3,51 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+type State = {
+  query: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    query: '',
+  };
+
+  searchHandler = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({
+      query: event.target.value,
+    });
+  };
 
   render() {
+    const { query } = this.state;
+    const loweredQuery = query.toLowerCase();
+    const visibleMovies = moviesFromServer
```

Comment:
create getVisibleMovies method
```suggestion
    const visibleMovies = this.getVisibleMovies(moviesFromServer);
```